src/components/TeacherComponents/OnboardingForm/onboardingActions/collectOnboardingTrackingProps.ts

```diff
@@ -0,0 +1,124 @@
+import { useUser } from "@clerk/nextjs";
+
+import { OnboardingFormProps } from "../OnboardingForm.schema";
+
+import {
+  onboardingRoleOptions,
+  OnboardingRoleValue,
+} from "./onboardingRoleOptions";
+
+import {
+  UserOnboardingCompletedProperties,
+  UserOnboardingProgressedProperties,
+  UserRoleTypeValueType,
+} from "@/browser-lib/avo/Avo";
+
+/* Reported for manually entered schools or as a fallback when the URN cannot be extracted
+from the school picker data */
+const DUMMY_URN = "000000";
+
+/**
+ * Produces onboarding tracking properties from the user and onboarding form
+ */
+export function collectOnboardingTrackingProps(
```

Comment:
Why must tracking require so much code be delivered to the client?!

![desmondwat](https://github.com/user-attachments/assets/c5adb21b-ef4b-46f4-bf81-e772bc31a434)

Reply:
For my understanding is this an Avo issue or something separate. I'm building up a bit of an "evaluation" of how useful Avo is as a middle-man for handling Posthog events.